addons/dynamic-mount/ossfs/docker/inotify.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+set -xe
+
+FUSE_CONFIG="/etc/fluid/config"
+
+mount_and_umount() {
+    mounted=$(cat /proc/self/mountinfo | grep fuse.ossfs | grep ${MOUNT_POINT} | awk '{print $5}')
+    python /mount_and_umount.py ${mounted}
+}
+
+# mount for init fuse-config
+mount_and_umount
+
+# trap SIGTERM, umount all
+trap "/umount.sh" SIGTERM SIGKILL
+
+# if fuse-config is modified, mount and umount to update
+# config.json is mounted by configmap, it is actually a symlink point to actual file,
+# and kubernetes would replace actual file by create new one and delete old one, so we handle inotify deletion event.
+# Please see https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/atomic_writer.go#L93-L138 for more information
+inotifywait -m -r -e delete_self "${FUSE_CONFIG}" |
```

Comment:
Here should be watch the `delete` event, not the `delete_self` event as the `${FUSE_CONFIG}` is a directory which will not be deleted when configmap changes.

Reply:
Thanks, I looked into `delete` and `delete_self` events, and here it could work because `-r` is used to watch the directory recursively. Such as for file `${FUSE_CONFIG}/..2023_11_14_05_33_00.762059865`, the event is `delete_self`. Do you think these two ways (`inotifywait -m -e delete "${FUSE_CONFIG}"` and `inotifywait -m -r -e delete_self "${FUSE_CONFIG}"`) should be carefully considered here?

Reply:
Sorry for confusing and the code is correct. The `-r delete_self` works when sub directory deleted. I was misled by `_self`.